oak-jcr/src/test/java/org/apache/jackrabbit/oak/jcr/ImportOperationsTest.java

```diff
@@ -0,0 +1,222 @@
+ * <p>
+ * Note that the purpose of these tests is not to check conformance with the JCR
+ * specification, but to observe the actual behavior of the implementation
+ * (which may be hard to change).
+ */
+public class ImportOperationsTest extends AbstractRepositoryTest {
+
+    private Session session;
+    private Node testNode;
+    private static String TEST_NODE_NAME = "ImportOperationsTest";
+    private static String TEST_NODE_NAME_REF = "ImportOperationsTest-Reference";
+    private static String TEST_NODE_NAME_TMP = "ImportOperationsTest-Temp";
+
+    public ImportOperationsTest(NodeStoreFixture fixture) {
+        super(fixture);
+    }
+
+    @Before
+    public void setup() throws RepositoryException {
+        this.session = getAdminSession();
+        this.testNode = session.getRootNode().addNode("import-tests", NodeType.NT_UNSTRUCTURED);
+        session.save();
+    }
+
+    @Test
+    public void jcrMixinCreatedOnNtUnstructured() throws RepositoryException {
+        Node test = testNode.addNode(TEST_NODE_NAME, NodeType.NT_UNSTRUCTURED);
+        try {
+            test.setProperty("jcr:created", false);
+            session.save();
+            test.addMixin(NodeType.MIX_CREATED);
+            session.save();
+            // surprise: setting the mixin succeeds, and the subsequent state of
+            // the node is inconsistent with the mixin's type definition
+            assertEquals(false, test.getProperty("jcr:created").getBoolean());
+        } finally {
+            test.remove();
+            session.save();
+        }
+    }
+
+    @Test
+    public void jcrMixinReferenceableOnNtUnstructuredBeforeSettingMixin() throws RepositoryException {
+        Node test = testNode.addNode(TEST_NODE_NAME, NodeType.NT_UNSTRUCTURED);
+        try {
+            String testUuid = UUID.randomUUID().toString();
+            test.setProperty("jcr:uuid", testUuid);
+            session.save();
+            test.addMixin(NodeType.MIX_REFERENCEABLE);
+            session.save();
+            // surprise: setting the mixin succeeds, and the subsequent state of
+            // the node doesn't have an autocreated UUID
+            assertEquals(testUuid, test.getProperty("jcr:uuid").getString());
+        } finally {
+            test.remove();
+            session.save();
+        }
+    }
+
+    @Test
+    public void jcrMixinReferenceableOnNtUnstructuredAfterSettingMixin() throws RepositoryException {
+        Node test = testNode.addNode(TEST_NODE_NAME, NodeType.NT_UNSTRUCTURED);
+        test.addMixin(NodeType.MIX_REFERENCEABLE);
+        session.save();
+        try {
+            String testUuid = UUID.randomUUID().toString();
+            test.setProperty("jcr:uuid", testUuid);
+            session.save();
+            fail("Setting jcr:uuid after adding mixin:referenceable should fail");
+        } catch (ConstraintViolationException ex) {
+            // expected
```

Comment:
This is IMHO mandated by the JCR spec as mix:referenceable defines jcr:uuid as protected (so this test rather belongs to the JCR TCK)

Reply:
If it's there we could drop it.